assets/js/settings.js

```diff
@@ -11,5 +44,28 @@
 				$email.removeAttr( 'required' );
 			}
 		} ).change();
+
+    // Code for reading uploaded papi file.
+		$( '#api_config_file' ).on( 'change', function (e) {
+			if (!e.target.files || !e.target.files[0]) {
+        return;
+      } else {
+        const file = e.target.files[0];
+        const reader = new FileReader();
+        reader.onload = function(f) {
+          // When a file is uploaded, the read contents populate the hidden textarea.
+          var contents = f.target.result;
+          updateCreds(contents);
```

Comment:
This is redundant - updating the input's contents should trigger the on change below which triggers `updateCreds`.

Reply:
I thought that too, but in testing, that initial FileReader function updated the input textarea but did not fire that subsequent on change, so I had to put in that weird redundancy.